cmd/main.go

```diff
@@ -136,21 +98,25 @@ func run() error {
 		connStr = "postgres://postgres:mysecretpassword@localhost:5432/test_db"
 	}
 	// give postgres the time to start when using docker compose
-	time.Sleep(5 * time.Second)
+	// time.Sleep(5 * time.Second)
 	err = postgres.Init(connStr)
 	if err != nil {
-		log.Fatalf("Error initializing database: %v", err)
+		return fmt.Errorf("error initializing database: %v", err)
```

Comment:
These returns stop the server from panicking when we encounter an error. log.Fatal will hard exit the server, we will never want to do this in a handlerfunc.